src/Tribe/Views/V2/View.php

```diff
@@ -1708,62 +1719,72 @@ protected function setup_template_vars() {
+			'after_events'                      => tribe( Advanced_Display::class )->get_after_events_html( $this ),
+			'display_events_bar'                => $this->filter_display_events_bar( $this->display_events_bar ),
 			/**
 			 * Allow filtering to determine whether or not to apply the `tribeDisableTribeBar` setting on the Events Manager page.
 			 *
 			 * @since 5.12.1
 			 */
-			'disable_event_search' => apply_filters( 'tec_events_views_v2_disable_tribe_bar', tribe_get_option( 'tribeDisableTribeBar', false ) ),
-			'live_refresh'         => tribe_is_truthy( 'automatic' === tribe_get_option( 'liveFiltersUpdate', 'automatic' ) ),
-			'ical'                 => $this->get_ical_data(),
-			'container_classes'    => $this->get_html_classes(),
-			'container_data'       => $this->get_container_data(),
-			'is_past'              => 'past' === $this->context->get( 'event_display_mode', false ),
-			'breakpoints'          => $this->get_breakpoints(),
-			'breakpoint_pointer'   => $this->get_breakpoint_pointer(),
-			'is_initial_load'      => $this->context->doing_php_initial_state(),
-			'public_views'         => $this->get_public_views( $url_event_date ),
-			'show_latest_past'     => $this->should_show_latest_past_events_view(),
-			'past'                 => $this->context->get( 'past', false ),
+			'disable_event_search'              => apply_filters( 'tec_events_views_v2_disable_tribe_bar', tribe_get_option( 'tribeDisableTribeBar', false ) ),
+			'live_refresh'                      => tribe_is_truthy( 'automatic' === tribe_get_option( 'liveFiltersUpdate', 'automatic' ) ),
+			'ical'                              => $this->get_ical_data(),
+			'container_classes'                 => $this->get_html_classes(),
+			'container_data'                    => $this->get_container_data(),
+			'is_past'                           => 'past' === $this->context->get( 'event_display_mode', false ),
+			'breakpoints'                       => $this->get_breakpoints(),
+			'breakpoint_pointer'                => $this->get_breakpoint_pointer(),
+			'is_initial_load'                   => $this->context->doing_php_initial_state(),
+			'public_views'                      => $this->get_public_views( $url_event_date ),
+			'show_latest_past'                  => $this->should_show_latest_past_events_view(),
+			'past'                              => $this->context->get( 'past', false ),
+			'category_colors_enabled'           => tribe( Category_Color_Dropdown_Provider::class )->should_display_on_view( self::get_template_slug() ),
+			'category_colors_category_dropdown' => tribe( Category_Color_Dropdown_Provider::class )->get_dropdown_categories(),
+			'category_colors_priority_category' => $category_colors_priority_category,
+			'category_colors_super_power'       => tribe_get_option( 'category-color-legend-superpowers', false ),
+			'category_colors_show_reset_button' => tribe_get_option( 'category-color-reset-button', false ),
```

Comment:
```suggestion
			'category_colors_enabled'           => tribe( Category_Color_Dropdown_Provider::class )->should_display_on_view( self::get_template_slug() ),
			'category_colors_category_dropdown' => tribe( Category_Color_Dropdown_Provider::class )->get_dropdown_categories(),
			'category_colors_priority_category' => $category_colors_priority_category,
			'category_colors_super_power'       => tribe_get_option( 'category-color-legend-superpowers', false ),
			'category_colors_show_reset_button' => tribe_get_option( 'category-color-reset-button', false ),
```
I will suggest that we inject the category colors related args using a filter. If there is none then add it in and then use that filter from the category colors controller to fill these.

We are putting features behind Controller contract to take advantage of the `is_active` method. This way if someone wants to disable a specific feature, that won't bloat the existing features.